services/slam/builtin/builtin.go

```diff
@@ -369,7 +371,32 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string) (*referencefr
 		return nil, errors.Wrap(err, "error getting SLAM position")
 	}
 
-	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
+	pInFrame := referenceframe.ProtobufToPoseInFrame(resp.Pose)
+
+	// TODO DATA-531: Remove extraction and conversion of quaternion from the extra field in the response once the Rust SDK
+	// is available and the desired math can be implemented on the orbSLAM side
+	extra := resp.Extra.AsMap()
```

Comment:
Will this panic if Extra is nil?

Reply:
No I have tested it without the extra line and it returns an empty interface